src/tls13.c

```diff
@@ -3199,7 +3199,12 @@ int DoTls13ServerHello(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
             ssl->version.minor = pv.minor;
         }
 
-        if (foundVersion && ssl->options.downgrade && (pv.minor < ssl->options.minDowngrade)) {
+        if (foundVersion && ssl->options.downgrade && 
+         /* Since found supported-version ex, this is TLSv1_3 server hello.
```

Comment:
Don't have comment in middle of if statements.
pv.minor will always be TLSv1_2_MINOR (line 3119)
ssl->options.minDowngrade can't be greater than TLSv1_3_MINOR.
Doesn't make sense to have this check at all.

What is the test case that the original statement was attempting to fix?

Reply:
Hello @SparkiDev 
The test case is setting TLS1_3_VERSION as minimum version thorough  wolfSSL_CTX_set_min_proto_version() before TLS connection:
 wolfSSL_CTX_set_min_proto_version(TLS1_3_VERSION) // minDowngrade will be TLSv1_3_MINOR
 SSL_connect()

In the test case above, the check will fail because pv.minor will be always TLSv1_2_MINOR as you mentioned, even though it uses TLSv 1.3.  But it should not be NG because it allows TLS v1.3 as minimum, I think.

Reply:
Your test case makes sense.
Why this statement was here at all doesn't make sense to me. Can you see a reason for having the original if statement (before you modified it) in there?

Reply:
I cannot find the reason why the check is in there too. It seems that removing the check makes sense, but i tried to keep it.

Reply:
I believe Tesfa made the change so ask him. Thanks.

Reply:
Oh, thanks. I understood what you asked at the beginning.

Reply:
Hi @tmael 
 Could you take a look at this change?
 I am seeing TLS connection failure due to the check when setting minDowngrade as TLSv1_3_MINOR. I cannot find any reason why the check is in there. So my change try to keep the check anyway. 
 However, it appears that removing the check makes sense. What is the test case that the original statement was attempting to fix?